Source/WebCore/rendering/svg/RenderSVGResourceClipper.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) Research In Motion Limited 2009-2010. All rights reserved.
+ * Copyright (C) 2021, 2022, 2023 Igalia S.L.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * along with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#pragma once
+
+#include "RenderSVGResourceContainer.h"
+#include "SVGGraphicsElement.h"
+#include "SVGUnitTypes.h"
+
+#include <wtf/HashMap.h>
+
+namespace WebCore {
+
+class GraphicsContext;
+class SVGClipPathElement;
+
+class RenderSVGResourceClipper final : public RenderSVGResourceContainer {
+    WTF_MAKE_ISO_ALLOCATED(RenderSVGResourceClipper);
+public:
+    RenderSVGResourceClipper(SVGClipPathElement&, RenderStyle&&);
+    virtual ~RenderSVGResourceClipper();
+
+    inline SVGClipPathElement& clipPathElement() const;
+
+    SVGGraphicsElement* shouldApplyPathClipping() const;
+    void applyPathClipping(GraphicsContext&, const FloatRect& objectBoundingBox, SVGGraphicsElement&);
+    void applyMaskClipping(PaintInfo&, const RenderLayerModelObject& targetRenderer, const FloatRect& objectBoundingBox);
+
+    void removeAllClientsFromCacheIfNeeded(bool, WeakHashSet<RenderObject>*) final { }
+    void removeClientFromCache(RenderElement&, bool = true) final { }
+
+    bool applyResource(RenderElement&, const RenderStyle&, GraphicsContext*&, OptionSet<RenderSVGResourceMode>) final { return true; }
+
+    FloatRect resourceBoundingBox(const RenderObject&, RepaintRectCalculation) final;
+
+    RenderSVGResourceType resourceType() const final { return ClipperResourceType; }
+
+    bool hitTestClipContent(const FloatRect&, const LayoutPoint&);
+
+    inline SVGUnitTypes::SVGUnitType clipPathUnits() const;
```

Comment:
Wrong, it's just not used consistently, there are places in RenderSVGResourceClipper that still use "clipPathElement().clipPathUnits()" instead of plain 'clipPathUnits()'.

Reply:
I removed those instances.